src/python/dynamic_queue.py

```diff
@@ -0,0 +1,66 @@
+class Node:
+    def __init__(self, key: int) -> None:
+        self.key = key
+        self.next = None
+
+
+class DynamicQueue:
+    def __init__(self) -> None:
+        self.__begin = None
+        self.__end = None
+
+    def is_empty(self) -> bool:
+        return self.__begin is None
+
+    def enqueue(self, val: int) -> None:
+        node = Node(val)
+        if self.is_empty():
+            self.__begin = node
+        else:
+            self.__end.next = node
+        self.__end = node
+
+    def dequeue(self) -> Node | None:
```

Comment:
just to be compatible to Python < 3.10
```suggestion
    def dequeue(self) -> Node:
```